scripts/js/commands/checkNewPills.ts

```diff
@@ -0,0 +1,107 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import fs from "fs/promises";
+import path from "path";
+
+import { globby } from "globby";
+import yargs from "yargs/yargs";
+import { hideBin } from "yargs/helpers";
+import { flatten, flattenDeep } from "lodash-es";
+
+import { TocEntry } from "../lib/api/generateToc.js";
+
+
+
+const TODAY = new Date();
+
+// Grab the toc and grab each url with a newpill
+async function main() {
+  const tocFiles = await globby([
+    "docs/guides/_toc.json",
+    "docs/migration-guides/_toc.json",
+    "docs/open-source/_toc.json"
+  ]);
+  const allOutdatedPills = [];
+  for (const tocFile of tocFiles) {
+    const tocNewPills = await findNewPills(tocFile);
+    const outdatedPills = await getOutdatedPills(tocNewPills);
+    if (outdatedPills.length > 0) {
+      allOutdatedPills.push(...outdatedPills);
+    }
+  }
+  if (allOutdatedPills.length > 0) {
+    console.error(
+      "\n❌ There are some outdated new pills! These URLs need to have their new pill removed: \n\n",
+      allOutdatedPills.map(item => item.join(' : ')).join("\n"),
+      "\n\n",
+    );
+  }
+  else {
+    console.log("\nNo outdated new pills found ✅\n");
+  }
+}
+
+async function getOutdatedPills(newPills: string[]): Promise<string[][]> {
+  const outdatedPills = [];
+  for (const pill of newPills) {
+    var isOld;
+    if (pill[1] == 'null') {
+      outdatedPills.push([pill[0], 'null']);
+    }
+    else{
+      const isOld = getDateDifference(pill[1]);
+      outdatedPills.push([pill[0], pill[1]]);
+    }
+  }
+  return outdatedPills;
+}
+
+function getDateDifference(pubDate: string): boolean {
+  const [year, month, day] = pubDate.split('-').map(Number);
+  const oldDate = new Date(year, month-1, day);
```

Comment:
The Date constructor understands our strings automatically.

```suggestion
  const oldDate = new Date(pubDate);
```